server-client/index.js

```diff
@@ -82,10 +82,17 @@ export class ServerClient {
     if (this.clientId) {
       this.app.clientIds.delete(this.clientId)
       this.app.nodeIds.delete(this.nodeId)
-      for (let i in this.app.subscribers) {
-        delete this.app.subscribers[i][this.nodeId]
-        if (Object.keys(this.app.subscribers[i]).length === 0) {
-          delete this.app.subscribers[i]
+      let clientNodeLog = new Log({
+        nodeId: this.nodeId,
+        store: new MemoryStore()
+      })
+      for (let channel in this.app.subscribers) {
+        let subscriber = this.app.subscribers[channel][this.nodeId]
+        if (subscriber) {
+          let action = { type: 'logux/unsubscribe', channel }
+          let actionId = clientNodeLog.generateId()
```

Comment:
It could generate an already used ID. You need to generate ID by `server.log.generateId()`, which leads to calling unsubscribe callback manually.

Reply:
Makes sense, done